api/integrations/google/gcal-bot

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import sys
+import time
+import datetime
+import optparse
+from six.moves import urllib
+import itertools
+import traceback
+import os
+import httplib2
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../'))
+import zulip
+from typing import List, Set, Tuple, Iterable, Optional
+
+from oauth2client import client, tools
+from oauth2client.file import Storage
+
+sys.path.append(os.path.dirname(__file__))
+
+parser = optparse.OptionParser(r"""
+
+%prog \
+    --user foo@zulip.com \
+    --calendar calendarID@example.calendar.google.com
+
+    This integration can be used to send yourself reminders, on Zulip, of Google Calendar Events.
+
+    Before running this integration make sure you run the get-google-credentials file to give Zulip
+    access to certain aspects of your Google Account.
+
+    This integration should be run on your local machine. Your API key and other information are
+    revealed to local users through the command line.
+
+    Depends on: google-api-python-client
+""")
+
+
+parser.add_option('--interval',
+                  dest='interval',
+                  default=30,
+                  type=int,
+                  action='store',
+                  help='Minutes before event for reminder [default: 30]',
+                  metavar='MINUTES')
+
+parser.add_option('--calendar',
+                  dest = 'calendarID',
+                  default = 'primary',
+                  type = str,
+                  action = 'store',
+                  help = 'Calendar ID for the calendar you want to receive reminders from.')
+
+parser.add_option_group(zulip.generate_option_group(parser))
+
+(options, args) = parser.parse_args()
+
+if not (options.zulip_email):
+    parser.error('You must specify --user')
+
+try:
+    from googleapiclient import discovery
+except ImportError:
+    parser.error('Install google-api-python-client')
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+CLIENT_SECRET_FILE = 'client_secret.json'
+APPLICATION_NAME = 'Zulip'
+
+def get_credentials():
+    # type: () -> client.Credentials
+    """Gets valid user credentials from storage.
+
+    If nothing has been stored, or if the stored credentials are invalid,
+    an exception is thrown and the user is informed to run the script in this directory to get
+    credentials.
+
+    Returns:
+        Credentials, the obtained credential.
+    """
+    try:
+        credential_dir = os.path.abspath(os.path.join(__file__, '../'))
+        credential_path = os.path.join(credential_dir,
+                                       'google-credentials.json')
```

Comment:
If you change the file name in the `get-google-credentials` script you have to change it here too. How about if it was a module constant in that module. But if you decide to go that way you have to make sure importing the module wouldn't run the `get_credentials` function in that module.